app_flutter/lib/commit_box.dart

```diff
@@ -5,29 +5,139 @@
 import 'package:flutter/material.dart';
 
 /// Displays Git commit information.
-class CommitBox extends StatelessWidget {
+///
+/// On click, it will open an [OverlayEntry] with [CommitOverlayContents]
+/// to show the information provided. Otherwise, it just shows the avatar
+/// for the author of this commit. Clicking outside of the [OverlayEntry]
+/// will close it.
+class CommitBox extends StatefulWidget {
   // TODO(chillers): convert to use commit model
-  const CommitBox({
-    Key key,
-    @required this.message,
-    @required this.avatarUrl,
-  }) : super(key: key);
+  const CommitBox(
+      {Key key,
+      @required this.message,
+      @required this.avatarUrl,
+      @required this.author})
+      : super(key: key);
 
   /// Commit message that summarizes the change made.
   final String message;
 
   /// Image URL to the avatar of the author of this commit.
   final String avatarUrl;
 
+  /// The person that authored this commit.
+  final String author;
+
+  @override
+  _CommitBoxState createState() => _CommitBoxState();
+}
+
+class _CommitBoxState extends State<CommitBox> {
+  OverlayEntry _commitOverlay;
+
   @override
   Widget build(BuildContext context) {
-    // TODO(chillers): add overlay to view more information
-    return Container(
-      margin: const EdgeInsets.all(1.0),
-      child: Image.network(
-        avatarUrl,
-        height: 40,
+    return GestureDetector(
+      onTap: _handleTap,
+      child: Container(
+        margin: const EdgeInsets.all(1.0),
+        child: Image.network(
+          widget.avatarUrl,
+          height: 40,
+        ),
       ),
     );
   }
+
+  void _handleTap() {
+    _commitOverlay = OverlayEntry(
+        builder: (nullContext) => CommitOverlayContents(
```

Comment:
nullContext is a bit weird of a name, I would name it `overlayContext` to be specific

Reply:
Thank you! My creativity had run out, but I knew it was null :)